includes/class-sensei-lesson.php

```diff
@@ -1118,7 +1118,11 @@ private function save_post_meta( $post_key = '', $post_id = 0 ) {
 
 		// Check if the user has permission to edit the target course.
 		if ( 'lesson_course' === $post_key && ! current_user_can( get_post_type_object( 'course' )->cap->edit_post, $new_meta_value ) ) {
-			return;
+			if ( empty( $new_meta_value ) ) {
```

Comment:
Although this kinda works, it does for the wrong reasons. Normally, this `if` checks if the user has the capability to edit the course and doesn't update the `lesson_course` meta if he doesn't. In this specific case, as `$new_meta_value` is empty for 'None' `current_user_can` will return false which seems a bit confusing.

I think it will be cleaner if we checked that `$new_meta_value` is empty in the outer if.

Reply:
Thanks @gkaragia, great feedback!  I've updated the PR with your suggestions :)